source/wp-content/themes/wporg-developer/inc/template-tags.php

```diff
@@ -1492,6 +1492,23 @@ function get_source_code( $post_id = null, $force_parse = false ) {
 			fclose( $handle );
 		}
 
+		// Trim leading whitespace.
+		if ( preg_match_all( "!^([\t ]*)(.*)$!m", $source_code, $m ) ) {
+			$non_empty_line_prefixes = array_filter(
+				$m[1],
+				function( $index ) use( $m ) {
+					return ! empty( $m[2][ $index ] );
+				},
+				ARRAY_FILTER_USE_KEY
+			);
+			if ( $non_empty_line_prefixes ) {
+				$strip_prefix = min( array_map( 'strlen', $non_empty_line_prefixes ) );
+				if ( $strip_prefix ) {
+					$source_code = preg_replace( "!^[\t ]{" . $strip_prefix . "}!m", '$1', $source_code );
+				}
+			}
```

Comment:
I had a dream.

Half of this code is simply to deal with the fact that the Regex is matching lines with code in them.. and empty lines alike.. I could just.. y'know.. have altered the regex to match what's needed...

```suggestion
		if ( preg_match_all( "!^([\t ]*).+$!m", $source_code, $m ) ) {
			$strip_prefix = min( array_map( 'strlen', $m[1]) );
			if ( $strip_prefix ) {
				$source_code = preg_replace( "!^[\t ]{" . $strip_prefix . "}!m", '$1', $source_code );
			}
```